recipes/sdl/all/test_package/CMakeLists.txt

```diff
@@ -1,13 +1,13 @@
 cmake_minimum_required(VERSION 3.15)
-project(test_package)
+project(test_package LANGUAGES C)
 
 find_package(SDL2 REQUIRED CONFIG)
 
-add_executable(${PROJECT_NAME} test_package.cpp)
-target_link_libraries(${PROJECT_NAME} SDL2::SDL2main)
+add_executable(${PROJECT_NAME} test_package.c)
+target_link_libraries(${PROJECT_NAME} PRIVATE SDL2::SDL2main)
```

Comment:
This causes a hard error on our end because we consume SDL with`-o:h sdl/*:sdl2main=False`.

It can also cause errors with `sdl2main=True` because the final test application will end up with two copies of `main()`. Since the test package doesn't use it, I would just remove this.

Reply:
We also needed
```python
self.cpp_info.components["libsdl2"].defines.append("SDL_MAIN_HANDLED")
```
It seems like the install target produces `SDL_main.h` headers regardless of the CMake settings. Might be an upstream issue though

Reply:
@SpaceIm Are you implementing this? Looks like a good solution to me

Reply:
It's out of scope of this PR. It can be done in another one.